sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -41,9 +42,25 @@ class ReflectiveActionRouter[A <: Action](action: A, componentMethods: Map[Strin
         componentMethod.requestMessageDescriptor,
         message.metadata())
 
-    componentMethod.method.get // safe call: if component method is None, proxy won't forward calls to it
-      .invoke(action, componentMethod.parameterExtractors.map(e => e.extract(context)): _*)
-      .asInstanceOf[Action.Effect[_]]
+    val inputTypeUrl =
+      message.payload().asInstanceOf[ScalaPbAny].typeUrl.replace(kalix.javasdk.JsonSupport.KALIX_JSON, "")
+
+    if (componentMethod.methodsMap.isEmpty)
+      componentMethod.method.get // safe call: if component is None the proxy won't forward calls to it
+        .invoke(action, componentMethod.parameterExtractors.map(e => e.extract(context)): _*)
+        .asInstanceOf[Action.Effect[_]]
+    else {
+      val method = Option(componentMethod.methodsMap.getOrElse(inputTypeUrl, null))
+      method.get // safe call: if component is None the proxy won't forward calls to it
+        .invoke(
+          action,
+          componentMethod.parameterExtractors.collect {
+            case e @ AnyBodyExtractor(cls)
+                if cls.getName.equals(inputTypeUrl) => //TODO This need to be tested thoroughly
+              e.extract(context)
```

Comment:
there are only two reasons we should get here. To discuss
   1. the AnyBodyExtractor doesn't match the argument of this method. That's expected when one have multiple subscription to an ES because each subscription has one event. TODO check that's not allowed.
  2. When there is another kind of extractor then componentMethod.methodsMap should be empty and
   it shouldn't reach here. When componentMethod.methodsMap has content it implies the rcp Input type is Any. This is an assumption. To verify.